pkg/template/app/app.go

```diff
@@ -41,25 +46,38 @@ type AppCROutput struct {
 
 func NewAppCR(config Config) ([]byte, error) {
 	userConfig := applicationv1alpha1.AppSpecUserConfig{}
-
-	var namespace string
+	appLabels := map[string]string{}
+
+	// Accomodating all the label cases here:
+	// 1. In-cluster Apps get unique label
+	// 2. Org-namespaced Apps get cluster label
+	// 3. Cluster-namespaced Apps with defaulting disabled gets default version label
+	// 4. Cluster-namespaced Apps with defaulting enabled gets nothing
+	var crNamespace string
 	if config.InCluster {
```

Comment:
Moved labels logic here, to have a single place to decide which ones to set. It seems cleaner to me, WDYT?